### ISSUE TITLE
Feature flag "autodie" mentioned in documentation does not exist

### ISSUE THREAD

**Issue by x-qq** [RexOps/Rex#1439]
## Describe the bug

The page https://metacpan.org/pod/Rex mentions "autodie" under feature flags, but this flag seems to no longer exist.

## How to reproduce it

1. try activating autodie flag
2. rex exits with error

Shortest code example that demonstrates the bug:

```bash

perl -E 'use Rex -feature => [qw[autodie]];'

```

## Expected behavior

If the documentation states that autodie feature is available, then it should be available.

## Circumstances

 - Rex version: 1.10.0
 - Perl version: 5.26.2
 - OS running rex: Debian Sid
 - OS managed by rex: not applicable
 - How rex was installed: cpanm Rex

## Debug log

```
% cat Rexfile
use Rex -feature => ['autodie'];
% rex -T -d  
[2020-10-06 12:34:57] DEBUG - This is Rex version: 1.10.0
[2020-10-06 12:34:57] DEBUG - Command Line Parameters
[2020-10-06 12:34:57] DEBUG - 	T = 1
[2020-10-06 12:34:57] DEBUG - 	d = 1
[2020-10-06 12:34:57] DEBUG - Found Rexfile.lock
[2020-10-06 12:34:57] DEBUG - Found stale lock file. Removing it.
[2020-10-06 12:34:57] DEBUG - Creating lock-file (Rexfile.lock)
[2020-10-06 12:34:57] DEBUG - Loading Rexfile
[2020-10-06 12:34:57] WARN - You tried to load a feature (autodie) that doesn't exists in your Rex version. Please update.
%
```

**Comment by nfg** [comment]
Hey! I took a look into this. As far as I can tell, the feature flags were documented in 598baaf1598beeeffff8488b9fd5a2a58f805b61 , but "autodie" never existed as an explicit feature flag -- it was just automatically enabled back in 0.56.

So... proposed fix: Just remove the documentation. Sound good?

**Comment by ferki** [comment]
@nfg: thanks for looking into this, and you're on the right track that the history should be examined to make a decision about the implementation :+1:

That being said, in the case of feature flags specifically, the [Contributing guide](https://github.com/RexOps/Rex/blob/1.13.0/CONTRIBUTING.md) documents the general desired approach:

> The goal is to have pairs of feature flags for opting in and out.

So for this issue I believe the solution is to have both an `autodie` and a `no_autodie` feature flag which are complementing each other (plus both are documented).

**Comment by ferki** [comment]
Based on a quick look, it seems that the interesting commit here is afc0ca1a4a8a880e8b7e00fe22f30a7152dce1ec, which has the message:

```
fixed #498  added autodie feature flag
```

but the changes on the commit doesn't really match that description. It doesn't contain an explicit `autodie` feature flag, only the related configuration option. But also sets a value for this config option if `0.56` or later versioned feature bundle is activated. To me this feels either a misleading commit message, or a non-complete implementation of the idea, but most probably it's simply an oversight.

Since the purpose of feature flags is to fine tune the behavior of rex, the user should be allowed to opt-in or opt-out to these settings to match their own use case at hand. Based on the reporter's example, there was a use case where an `autodie` feature was desired without any other features (including versioned bundles).

All in all I think it would be best to make sure that an opt-in `autodie` feature flag is present as a complementer to the already existing opt-out `no_autodie`.

**Comment by ferki** [comment]
@xq: if you'd need a workaround for the lack of explicit `autodie` feature while this is getting resolved, I believe this would help:

```perl
Rex::Config->set_autodie(1);
```
